test/discounts/TBADiscountValidator/IsValidDiscountRegistration.t.sol

```diff
@@ -0,0 +1,30 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {TBADiscountValidator} from "src/L2/discounts/TBADiscountValidator.sol";
+import {TBADiscountValidatorBase} from "./TBADiscountValidatorBase.t.sol";
+
+contract IsValidDiscountRegistration is TBADiscountValidatorBase {
+    function test_reverts_whenTheSignatureIsExpired() public {
+        bytes memory validationData = _getDefaultValidationData();
+        (bytes32 _deviceId,, bytes memory sig) = abi.decode(validationData, (bytes32, uint64, bytes));
+        bytes memory expiredSignatureData = abi.encode(_deviceId, (block.timestamp - 1), sig);
```

Comment:
I wonder whether we should do something like this:
```solidity
        bytes memory expiredSignatureData = _getDefaultValidationData();
        vm.warp(expires+1);
```

The signature data that's provided no longer matches the signed message so the ecrecover will return a different address to the signer. Given that we check the expiry before doing the ecrecover, it doesn't make a practical difference

Reply:
Good call!